social_django/views.py

```diff
@@ -39,21 +43,88 @@ def disconnect(request, backend, association_id=None):
                          redirect_name=REDIRECT_FIELD_NAME)
 
 
+def get_session_timeout(
+        social_user,
+        enable_session_expiration=False,
+        max_session_length=None,
+):
+    if enable_session_expiration:
+        # Retrieve an expiration date from the social user who just finished
+        # logging in; this value was set by the social auth backend, and was
+        # typically received from the server.
+        expiration = social_user.expiration_datetime()
+
+        # We've enabled session expiration. Check to see if we got
+        # a specific expiration time from the provider for this user;
+        # if not, use the platform default expiration.
+        received_expiration_time = (
+            expiration.seconds + (expiration.days * 86400)
+        ) if expiration else DEFAULT_SESSION_TIMEOUT
```

Comment:
Sure! It turns out that timedelta has [a `total_seconds()` method](https://docs.python.org/3.3/library/datetime.html#datetime.timedelta.total_seconds), so I'm swapping that in to fit the 80-character line (and also, save a few bytes).